src/Workspaces/Remote/Core/BrokeredServiceConnection.cs

```diff
@@ -24,33 +21,78 @@ namespace Microsoft.CodeAnalysis.Remote
     internal sealed class BrokeredServiceConnection<TService> : RemoteServiceConnection<TService>
         where TService : class
     {
+        private readonly struct Rental : IDisposable
```

Comment:
❔ Can we not just use `ServiceBrokerClient.Rental<TService>` directly?

Reply:
This is to minimize duplication of code across all variants of TryInvokeAsync.